src/Workspaces/CSharp/Portable/Formatting/Rules/SpacingFormattingRule.cs

```diff
@@ -174,22 +176,12 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
             {
                 return AdjustSpacesOperationZeroOrOne(optionSet, CSharpFormattingOptions.SpaceWithinSquareBrackets);
             }
-            else if (currentKind == SyntaxKind.CloseBracketToken && HasFormattableBracketParent(currentToken))
+            if (currentKind == SyntaxKind.CloseBracketToken && previousKind != SyntaxKind.OmittedArraySizeExpressionToken && HasFormattableBracketParent(currentToken))
             {
-                if (currentToken.Parent is ArrayRankSpecifierSyntax)
-                {
-                    var parent = currentToken.Parent as ArrayRankSpecifierSyntax;
-                    if ((parent.Sizes.Any() && parent.Sizes.First().Kind() != SyntaxKind.OmittedArraySizeExpression) || parent.Sizes.SeparatorCount > 0)
-                    {
-                        // int []: added spacing operation on open [
-                        // int[1], int[,]: need spacing operation
-                        return AdjustSpacesOperationZeroOrOne(optionSet, CSharpFormattingOptions.SpaceWithinSquareBrackets);
-                    }
-                }
-                else
-                {
-                    return AdjustSpacesOperationZeroOrOne(optionSet, CSharpFormattingOptions.SpaceWithinSquareBrackets);
-                }
+                // int []: added spacing operation on [
```

Comment:
Usually we add the case that gets handled by the `if` block next to the `if` block. So initially, I got confused by reading the comments that all 3 cases were handled by this block. 

For clarity, could you please add the first 2 comments next to the appropriate `if` block that handles these cases. Modify this comment to something like. 
```C#
\\ The following 2 cases have been handled earlier\later. It is *NOT* handled here
\\ case1: int []
\\ case2: int[,]

\\ The cases handled here are:
\\ case1: int[1]
```